tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
-    yum_repo = entities.Repository(
-        name=f'{product_name}_repo', product=product, url=repo_url, content_type='yum'
+@pytest.fixture(scope='module')
+def pre_upgrade_cv(default_org, pre_upgrade_repo):
+    """Create and publish repo"""
+    pre_upgrade_cv = entities.ContentView(
+        organization=default_org, repository=[pre_upgrade_repo.id]
     ).create()
-    yum_repo.sync()
-    return product, yum_repo
+    pre_upgrade_cv.publish()
+    pre_upgrade_cv = pre_upgrade_cv.read()
+    return pre_upgrade_cv
 
 
-def update_product_subscription_in_ak(product, yum_repo, ak, org):
-    """Updates given products subscription in given AK
-
-    :param nailgun.entities.Product product: products name to calculate
-        subscription id
-    :param nailgun.entities.Repository yum_repo: yum repository
-    :param nailgun.entities.ActivationKey ak: Ak
-    :param nailgun.entities.Organization org: Organization
-    """
-    cv_from_ak = ak.content_view
-    cv = cv_from_ak.read()
-    cv.repository.append(yum_repo)
-    cv = cv.update(['repository'])
-    cv.publish()
-    cv = cv.read()  # Published CV with new version
-    # Promote CV
-    environment = (
-        entities.ActivationKey(organization=org)
-        .search(query={'search': f'name={ak.name}'})[0]
-        .environment
+@pytest.fixture(scope='module')
+def post_upgrade_cv(default_org, post_upgrade_repo):
+    """Create and publish repo"""
+    post_upgrade_cv = entities.ContentView(
+        organization=default_org, repository=[post_upgrade_repo.id]
+    ).create()
+    post_upgrade_cv.publish()
+    post_upgrade_cv = post_upgrade_cv.read()
+    return post_upgrade_cv
+
+
+@pytest.fixture(scope='module')
+def module_ak(default_org, module_lce_library, pre_upgrade_repo):
+    """Create a module AK in Library LCE"""
+    ak = entities.ActivationKey(
+        content_view=DEFAULT_CV,
+        environment=module_lce_library,
+        organization=default_org,
+    ).create()
+    # Fetch available subscriptions
+    subs = entities.Subscription(organization=default_org).search()
+    assert len(subs) > 0
+    # Add default subscription to activation key
+    sub_found = False
+    for sub in subs:
+        if sub.name == DEFAULT_SUBSCRIPTION_NAME:
+            ak.add_subscriptions(data={'subscription_id': sub.id})
+            sub_found = True
+    assert sub_found
+    # Enable RHEL tools repo in activation key
+    ak.content_override(
+        data={'content_overrides': [{'content_label': REPOS['rhst7']['id'], 'value': '1'}]}
     )
-    cvv = entities.ContentViewVersion(id=max([cvv.id for cvv in cv.version])).read()
-    cvv.promote(data={'environment_id': environment.id, 'force': False})
-    subscription = entities.Subscription(organization=org).search(
+    # Add custom subscription to activation key
+    prod = pre_upgrade_repo.product.read()
+    custom_sub = entities.Subscription().search(query={'search': f'name={prod.name}'})
```

Comment:
```suggestion
    custom_sub = default_subscription.read()
```

Reply:
Its not default sub, its custom sub. I have changed it to:   

 `# Add custom subscription to activation key
    custom_sub = entities.Subscription().search(query={'search': f'name={module_product.name}'})`